webapp/src/components/shared/TokenUsageGraph.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+    Body1,
+    Body1Strong,
+    Body2,
+    Button,
+    Divider,
+    makeStyles,
+    Popover,
+    PopoverSurface,
+    PopoverTrigger,
+    shorthands,
+    Text,
+    tokens,
+} from '@fluentui/react-components';
+import { useAppSelector } from '../../redux/app/hooks';
+import { RootState } from '../../redux/app/store';
+import { TypingIndicator } from '../chat/typing-indicator/TypingIndicator';
+import { Info16 } from './BundledIcons';
+
+const useClasses = makeStyles({
+    horizontal: {
+        display: 'flex',
+        ...shorthands.gap(tokens.spacingVerticalSNudge),
+        alignItems: 'center',
+    },
+    content: {
+        display: 'flex',
+        flexDirection: 'column',
+        ...shorthands.gap(tokens.spacingHorizontalS),
+        paddingBottom: tokens.spacingHorizontalM,
+    },
+    popover: {
+        width: '300px',
+    },
+    header: {
+        marginBlockEnd: tokens.spacingHorizontalM,
+    },
+});
+
+interface ITokenUsageGraph {
+    promptUsage?: number;
+    dependencyUsage?: number;
+    promptView?: boolean;
+}
+
+export const TokenUsageGraph: React.FC<ITokenUsageGraph> = ({ promptView, promptUsage, dependencyUsage }) => {
+    const { conversations, selectedId } = useAppSelector((state: RootState) => state.conversations);
+
+    const hasTokenCount = (promptUsage ?? dependencyUsage) !== undefined;
+    const loadingResponse = conversations[selectedId].botResponseStatus;
+
+    // Necessary conversion due to type coercion issues
+    promptUsage = Number(promptUsage);
+    dependencyUsage = Number(dependencyUsage);
+
+    const classes = useClasses();
+    const maxWidth = 500;
```

Comment:
nit: use a constant

Reply:
It is a constant, it's a style that only applies to this component so not need to add it at a larger scale